### PR TITLE
Add more checks for search offset

### PR DESCRIPTION
**Pull request by XuanYang-cn**
See also: #19250

Signed-off-by: yangxuan <xuan.yang@zilliz.com>

/kind bug

### CODE REVIEW

File: internal/proxy/util.go [hunk 1]
```diff
@@ -74,10 +74,10 @@ func isNumber(c uint8) bool {
 	return true
 }
 
-func validateTopK(topK int64) error {
+func validateLimit(limit int64) error {
 	// TODO make this configurable
-	if topK <= 0 || topK >= 16385 {
-		return fmt.Errorf("limit should be in range [1, 16385], but got %d", topK)
+	if limit <= 0 || limit >= 16385 {
+		return fmt.Errorf("should be in range [1, 16385], but got %d", limit)
```
Comment (jiaoew1991) [inline internal/proxy/util.go:80]: range is [1, 16384]

Comment (XuanYang-cn) [reply internal/proxy/util.go:80]: so >= 16385 is invalid

Comment (congqixia) [reply internal/proxy/util.go:80]: What's the default value for limit if user does not provide the limit param?

**Pull request merged**